src/v2/Apps/Fair/routes.tsx

```diff
@@ -0,0 +1,22 @@
+import loadable from "@loadable/component"
+import { graphql } from "react-relay"
+import { RouteConfig } from "found"
+
+const FairApp = loadable(() => import("./FairApp"))
+
+export const routes: RouteConfig[] = [
+  {
+    path: "/fair2/:slug",
+    getComponent: () => FairApp,
+    prepare: () => {
+      FairApp.preload()
+    },
+    query: graphql`
+      query routes_FairQuery($slug: String!) {
+        fair(id: $slug) {
```

Comment:
could use `@principalField` on this field. Then you might not need https://github.com/artsy/force/pull/6117/files#diff-e0e09f09159fe3e65e3c76174b72b096R16 (although that's probably a good idea anyway?), you'd get the automatic 404 (or any other HTTP code if Gravity returns it) behavior.